extensions-contrib/parquet-extensions/src/main/java/io/druid/data/input/parquet/ParquetHadoopInputRowParser.java

```diff
@@ -56,15 +66,46 @@ public ParquetHadoopInputRowParser(
     }
   }
 
+
+  private LogicalType determineTimestampSpecLogicalType(Schema schema, String timestampSpecField)
+  {
+    for (Schema.Field field : schema.getFields()) {
+      if (field.name().equals(timestampSpecField)) {
+        this.timestampSpecLogicalType = field.schema().getLogicalType();
+        return this.timestampSpecLogicalType;
+      }
+    }
+    // The timestamp field does not exist in the provided schema
+    return null;
+  }
+
   /**
    * imitate avro extension {@link AvroStreamInputRowParser#parseGenericRecord(GenericRecord, ParseSpec, List, boolean, boolean)}
    */
   @Override
   public InputRow parse(GenericRecord record)
   {
+    // We don't want to determine the type every time if we already done so
+    LogicalType logicalType = this.timestampSpecLogicalType;
+    if (null == logicalType) {
+      logicalType = determineTimestampSpecLogicalType(record.getSchema(), timestampSpec.getTimestampColumn());
```

Comment:
For a single dataset the column type should be all the same. Could be that your time dimension evolves over time, but this is not supported by Druid anyway because for each datasource there is a single format.

I changed it from a static to a non-static value, because this caused the tests to fail (thanks tests). If you first parse a date, and then a string, this will get ugly because the string will be parsed as a date.

The idea of the caching is from here:
https://github.com/druid-io/druid/blob/63a897c278bb46702345b2f08b0b7b400d0bb925/api/src/main/java/io/druid/data/input/impl/TimestampSpec.java#L101

But as long as you don't have very wide parquet files, which you shouldn't, the overhead will not be a problem. 

Regarding the concurrency question, I expect the `parse(record)` to be called from a single thread from a single mapper.